File: pascal_triangle.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<vector<int>> generate(int numRows) {
    if (numRows <= 0) return {};

    vector<vector<int>> triangle(numRows);
    for (int i = 0; i < numRows; ++i) {
        triangle[i] = vector<int>(i + 1, 1);
        for (int j = 1; j < i; ++j) {
            triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
        }
    }
    return triangle;
}

int main() {
    int numRows = 5;
    auto result = generate(numRows);
    for (const auto& row : result) {
        cout << "[";
        for (size_t i = 0; i < row.size(); ++i) {
            cout << row[i];
            if (i < row.size() - 1) cout << ", ";
        }
        cout << "]\n";
    }
    return 0;
}

File: pascal_triangle.py

from typing import List


def generate(numRows: int) -> List[List[int]]:
    
    if numRows <= 0:
        return []

    triangle = [[1] * (i + 1) for i in range(numRows)]

    for i in range(2, numRows):
        for j in range(1, i):
            triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j]

    return triangle

result=generate(5)  # Example usage
print(result)  # Output: [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1], [1, 4, 6, 4, 1]]